lib/api/hub_config_api.rb

```diff
@@ -1,20 +1,23 @@
 class HubConfigApi
+  include HubEnvironmentConcern
   require 'cgi'
-
-  HUB_HOST = Rails.configuration.hub_config_host
   HEALTHCHECK_ENDPOINT = 'service-status'.freeze
   CERTIFICATES_ROUTE = '/config/certificates/'.freeze
   CERTIFICATE_ENCRYPTION_ENDPOINT = "%{entity_id}/certs/encryption".freeze
   CERTIFICATES_SIGNING_ENDPOINT = "%{entity_id}/certs/signing".freeze
 
-  def initialize; end
+  def initialize(environment:)
```

Comment:
you could add the line `attr_reader :hub_host, :environment, :authentication_header` above and then you wouldn't need to use `@` instance variables for these values outside of the initialiser.

Reply:
also, do we need to have the enviroment in the initializer? it means we'll have to create a new instance of  HubConfigApi everytime we want to make a call. Instead we could create a single instance on startup and use it with different URLs...

Reply:
I know I could do this, I didn't want to expose these attributes to the consumers of HubConfigApi. I suppose since they are read-only it does hurt. I'll change this since you guys prefer it.